spec/firebase_app_distribution_action_spec.rb

```diff
@@ -28,28 +29,112 @@
   end
 
   describe '#get_value_from_value_or_file' do
-    it 'should return the release notes string' do
-      expect(Fastlane::Actions::FirebaseAppDistributionAction.get_value_from_value_or_file("Hello World", "")).to eq("Hello World")
+    it 'returns the value when defined and the file path is empty' do
+      expect(action.get_value_from_value_or_file("Hello World", "")).to eq("Hello World")
     end
 
-    it 'should return the release notes from file' do
+    it 'returns the value when value is defined and the file path is nil' do
+      expect(action.get_value_from_value_or_file("Hello World", nil)).to eq("Hello World")
+    end
+
+    it 'returns the release notes when the file path is valid and value is not defined' do
+      expect(File).to receive(:open)
+        .with("file_path")
+        .and_return(fake_binary)
+      expect(action.get_value_from_value_or_file("", "file_path")).to eq("Hello World")
+    end
+
+    it 'returns the release notes when the file ath is valid and value is nil ' do
       expect(File).to receive(:open)
-        .with("release_notes_path")
+        .with("file_path")
         .and_return(fake_binary)
-      expect(Fastlane::Actions::FirebaseAppDistributionAction.get_value_from_value_or_file("", "release_notes_path")).to eq("Hello World")
+      expect(action.get_value_from_value_or_file(nil, "file_path")).to eq("Hello World")
     end
 
-    it 'should raise an error due to invalid release notes path ' do
+    it 'raises an error when an invalid path is given and value is not defined' do
       expect(File).to receive(:open)
-        .with("invalid_release_notes_path")
+        .with("invalid_path")
         .and_raise(Errno::ENOENT.new("file not found"))
-      expect { Fastlane::Actions::FirebaseAppDistributionAction.get_value_from_value_or_file("", "invalid_release_notes_path") }
-        .to raise_error("#{ErrorMessage::INVALID_PATH}: invalid_release_notes_path")
+      expect { action.get_value_from_value_or_file("", "invalid_path") }
+        .to raise_error("#{ErrorMessage::INVALID_PATH}: invalid_path")
+    end
+
+    it 'raises an error when an invalid path is given and value is nil' do
+      expect(File).to receive(:open)
+        .with("invalid_path")
+        .and_raise(Errno::ENOENT.new("file not found"))
+      expect { action.get_value_from_value_or_file(nil, "invalid_path") }
+        .to raise_error("#{ErrorMessage::INVALID_PATH}: invalid_path")
+    end
+  end
+
+  describe '#string_to_array' do
+    it 'returns an array when a string is passed in with no commas' do
+      array = Fastlane::Actions::FirebaseAppDistributionAction.string_to_array("string")
```

Comment:
Now that you have the `action` variable, please use it everywhere for consistency.

Reply:
That was the plan, but since we were still refactoring I was unsure if it was worth making that change. Given the initial refactor PR then it makes sense to change it.